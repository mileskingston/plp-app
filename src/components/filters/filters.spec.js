import React from 'react';
import renderer from 'react-test-renderer';
import Filters from './filters';

describe('Filters Component', () => {
  const facets = [{
    value: 'value 1',
    name: 'name 1',
    entry: [{
      count: 2,
      label: 'entry label 1',
      value: 'entry 1',
    }, {
      count: 1,
      label: 'entry label 2',
      value: 'entry 2',
    }],
  },
  {
    value: 'value 2',
    name: 'name 2',
    entry: [{
      count: 2,
      label: 'entry label 1',
      value: 'entry 1',
    }, {
      count: 1,
      label: 'entry label 2',
      value: 'entry 2',
    }],
  }];

  it('renders component', () => {
    let component = renderer.create(<Filters facets={facets} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});