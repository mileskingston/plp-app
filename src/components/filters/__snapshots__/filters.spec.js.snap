// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Component renders component 1`] = `
<div
  className="filters"
>
  <button
    className="button filters__button"
    onClick={[Function]}
    type="button"
  >
    Filters
  </button>
  <div
    className="filters-facets filters-facets--hidden"
  >
    <div
      className="facet"
    >
      <h3>
        name 1
      </h3>
      <label
        className="facet-entry"
        ufor="entry label 1"
      >
        <input
          className="facet-entry__input"
          id="entry label 1"
          name="entry label 1"
          tabIndex="0"
          type="checkbox"
          value="entry 1"
        />
        <span
          className="facet-entry__label"
        >
          entry label 1
           
          <span>
            2
          </span>
        </span>
      </label>
      <label
        className="facet-entry"
        ufor="entry label 2"
      >
        <input
          className="facet-entry__input"
          id="entry label 2"
          name="entry label 2"
          tabIndex="0"
          type="checkbox"
          value="entry 2"
        />
        <span
          className="facet-entry__label"
        >
          entry label 2
           
          <span>
            1
          </span>
        </span>
      </label>
    </div>
    <div
      className="facet"
    >
      <h3>
        name 2
      </h3>
      <label
        className="facet-entry"
        ufor="entry label 1"
      >
        <input
          className="facet-entry__input"
          id="entry label 1"
          name="entry label 1"
          tabIndex="0"
          type="checkbox"
          value="entry 1"
        />
        <span
          className="facet-entry__label"
        >
          entry label 1
           
          <span>
            2
          </span>
        </span>
      </label>
      <label
        className="facet-entry"
        ufor="entry label 2"
      >
        <input
          className="facet-entry__input"
          id="entry label 2"
          name="entry label 2"
          tabIndex="0"
          type="checkbox"
          value="entry 2"
        />
        <span
          className="facet-entry__label"
        >
          entry label 2
           
          <span>
            1
          </span>
        </span>
      </label>
    </div>
  </div>
</div>
`;
